### DISCUSSION TITLE
Hiding code that was not hit in the editor

### DISCUSSION THREAD

**Discussion by jcmorrow** [RecordReplay/devtools#4153] [Ideas]
This is something that I would love when I'm debugging library code, where I suspect that a particular bug is happening in a particular file, but I don't know that file super well. Being able to look through *only* the parts of that file that actually *ran* during this replay, would make it much easier to narrow down where I should be placing my breakpoint. I don't know if CodeMirror has folds, but it seems like a natural way to do this might be folds similar to what you see in github commit summaries.

**Comment by jonbell-lot23** [comment]
100%. I've been thinking about this for a whole year. Sometimes I know the codebase really well. But other times I'm overwhelmed and confused and I need help finding exactly this sort of data.

**Comment by jonbell-lot23** [comment]
@jcmorrow we discussed this in product office hours last week, I believe.

Where did we net out? I think I recalled that this was doable? And I think I saw someone say we're working on it?

**Reply by jcmorrow** [reply]
I have been working on it! It was a bit of a stretch for me and required me to learn a bunch of new things, so it took a bit longer than expected, and there are still 1 or 2 things that I need to clean up about it (though there are PR's 
[here](https://github.com/RecordReplay/devtools/pull/5788) and [here](https://github.com/RecordReplay/backend/pull/4899). I've been calling the feature Live Code Analysis [credit to Dan for the name]).

**Reply by markerikson** [reply]
given my first two days on this product, I suddenly _really_ want this feature _right now_ :)

**Reply by jonbell-lot23** [reply]
(nod nod) Excellent. I'm excited that we're to the point where the burden of proof is on why a person *wouldn't* want this rather than why they would. If it's too hard in the code, that's one thing. If it breaks something else, that's a fair reason not to do it.

But it sounds like we can do it, which I think it's going to be a really big deal for the sorts of debugging I find myself doing. And apparently I'm not alone!

**Reply by jcmorrow** [reply]
I think it has a lot to offer in the "dive into a foreign code base" use-cases for Replay.

**Reply by jonbell-lot23** [reply]
And when you're not great at debugging, everything (even your own project) is a foreign codebase 😂
